spec/unit/pops/types/p_binary_type_spec.rb

```diff
@@ -0,0 +1,218 @@
+require 'spec_helper'
+require 'puppet/pops'
+require 'puppet_spec/compiler'
+
+module Puppet::Pops
+module Types
+describe 'Binary Type' do
+  include PuppetSpec::Compiler
+
+  context 'as a type' do
+    it 'can be created with the type factory' do
+      t = TypeFactory.binary()
+      expect(t).to be_a(PBinaryType)
+      expect(t).to eql(PBinaryType::DEFAULT)
+    end
+
+    context 'when used in Puppet expressions' do
+
+        it 'the Binary type is equal to itself only' do
+          code = <<-CODE
+            $t = Binary
+            notice(Binary =~ Type[ Binary ])
+            notice(Binary == Binary)
+            notice(Binary < Binary)
+            notice(Binary > Binary)
+          CODE
+          expect(eval_and_collect_notices(code)).to eql(['true', 'true', 'false', 'false'])
```

Comment:
This should be replaced by adding the Binary type in the types_setup.rb - not meaningful to have another place to check assignability against all other types.
The test here (that you commented on) is quite meaningless. I am removing it.